hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -449,9 +450,32 @@ private Constants() {
    * May be set within a JCEKS file.
    * Value: "{@value}".
    */
+  @Deprecated
   public static final String SERVER_SIDE_ENCRYPTION_KEY =
       "fs.s3a.server-side-encryption.key";
 
+  /**
+   * Set S3-server side encryption(SSE) or S3-Client side encryption(CSE)
+   * algorithm. Check {@link S3AEncryptionMethods} for valid options.
+   * <br>
+   * value: {@value}
+   */
+  public static final String S3_ENCRYPTION_ALGORITHM =
+      "fs.s3a.encryption-algorithm";
```

Comment:
I think we should call this "fs.s3a.encryption.algorithm" for consistency with the key. 
It's a bigger change than before, but we could also add a deprecation entry for fs.s3a.encryption-algorithm in case people got it wrong